cmd/soroban-rpc/internal/test/cli_test.go

```diff
@@ -50,25 +51,37 @@ func outputsContractIDInLastLine(t *testing.T, output string) {
 
 func TestCLIContractDeploy(t *testing.T) {
 	NewCLITest(t)
-	output := runSuccessfulCLICmd(t, "contract deploy --salt 0 --wasm "+helloWorldContractPath)
+	output := deploy(t, helloWorldContractPath, 0)
 	outputsContractIDInLastLine(t, output)
 }
 
 func TestCLIContractDeployAndInvoke(t *testing.T) {
 	NewCLITest(t)
-	output := runSuccessfulCLICmd(t, "contract deploy --salt=0 --wasm "+helloWorldContractPath)
-	contractID := strings.TrimSpace(output)
-	output = runSuccessfulCLICmd(t, fmt.Sprintf("contract invoke --id %s -- hello --world=world", contractID))
+	contractID := runSuccessfulCLICmd(t, "contract deploy --salt=0 --wasm "+helloWorldContractPath)
+	output := runSuccessfulCLICmd(t, fmt.Sprintf("contract invoke --id %s -- hello --world=world", contractID))
 	require.Contains(t, output, `["Hello","world"]`)
 }
 
+func TestCLISimulateTransactionRestoreFromPreambleInFootprint(t *testing.T) {
+	NewCLITest(t)
+	strkeyContractID := runSuccessfulCLICmd(t, "contract deploy --salt=0 --wasm "+helloWorldContractPath)
+	count := runSuccessfulCLICmd(t, fmt.Sprintf("contract invoke --id %s -- inc", strkeyContractID))
+	require.Equal(t, "1", count)
+	count = runSuccessfulCLICmd(t, fmt.Sprintf("contract invoke --id %s -- inc", strkeyContractID))
+	require.Equal(t, "2", count)
+	time.Sleep(time.Second * 60)
```

Comment:
Can you wait for the ledger entry to expire? (with Go)

Reply:
I've tried several different ways and all result it it only reading account keys and not the ledger key.

Reply:
Let me try